src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -410,26 +433,11 @@ private static int GetPositionWithinDocumentBounds(int position, int documentLen
         private static TextSpan GetSpanWithinDocumentBounds(TextSpan span, int documentLength)
             => TextSpan.FromBounds(GetPositionWithinDocumentBounds(span.Start, documentLength), GetPositionWithinDocumentBounds(span.End, documentLength));
 
-        private static Document? OpenDocument(Workspace workspace, DocumentId documentId)
-        {
-            // Always open the document again, even if the document is already open in the 
-            // workspace. If a document is already open in a preview tab and it is opened again 
-            // in a permanent tab, this allows the document to transition to the new state.
-            if (workspace.CanOpenDocuments)
-            {
-                workspace.OpenDocument(documentId);
-            }
-
-            if (!workspace.IsDocumentOpen(documentId))
-            {
-                return null;
-            }
-
-            return workspace.CurrentSolution.GetDocument(documentId);
-        }
-
-        public bool NavigateTo(ITextBuffer textBuffer, VsTextSpan vsTextSpan, CancellationToken cancellationToken)
+        public async Task<bool> NavigateToTextBufferAsync(
+            ITextBuffer textBuffer, VsTextSpan vsTextSpan, CancellationToken cancellationToken)
         {
+            Contract.ThrowIfNull(textBuffer);
```

Comment:
Is this file and callers not annotated?

Reply:
callers are not annotated.  so i'm being hyper anal.